.github/workflows/publish.yml

```diff
@@ -0,0 +1,159 @@
+name: Labelbox Python SDK Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Release Tag'
+        required: true
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.ACTIONS_ACCESS_TOKEN  }}
+          ref: ${{ inputs.tag }}
+      - name: Install the latest version of rye
+        uses: eifinger/setup-rye@v2
+        with:
+          version: '0.29.0'
+          enable-cache: true
+      - name: Rye Setup
+        run: |
+           rye config --set-bool behavior.use-uv=true
+      - name: Create build
+        working-directory: libs/labelbox
+        run: |
+          rye sync
+          rye build
+      - uses: actions/upload-artifact@v4
+        with:
+          name: build
+          path: ./dist
+  test-build:
+    needs: ['build']
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - python-version: 3.8
+            prod-key: PROD_LABELBOX_API_KEY_2
+            da-test-key: DA_GCP_LABELBOX_API_KEY
+          - python-version: 3.9
+            prod-key: PROD_LABELBOX_API_KEY_3
+            da-test-key: DA_GCP_LABELBOX_API_KEY
+          - python-version: "3.10"
+            prod-key: PROD_LABELBOX_API_KEY_4
+            da-test-key: DA_GCP_LABELBOX_API_KEY
+          - python-version: 3.11
+            prod-key: LABELBOX_API_KEY
+            da-test-key: DA_GCP_LABELBOX_API_KEY
+          - python-version: 3.12
+            prod-key: PROD_LABELBOX_API_KEY_5
+            da-test-key: DA_GCP_LABELBOX_API_KEY
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.ACTIONS_ACCESS_TOKEN  }}
+          ref: ${{ inputs.tag }}
+      - name: Install the latest version of rye
+        uses: eifinger/setup-rye@v2
+        with:
+          version: '0.29.0'
+          enable-cache: true
+      - name: Rye Setup
+        run: |
+           rye config --set-bool behavior.use-uv=true
+      - name: Python setup
+        run: rye pin ${{ matrix.python-version }}
+      - uses: actions/download-artifact@v4
+        with:
+          name: build
+          path: ./dist
+      - name: Prepare package and environment
+        run: |
+          rye sync
+          rye run toml unset --toml-path pyproject.toml tool.rye.workspace
+          rye sync -f --update-all
+      - name: Integration Testing
+        env:
```

Comment:
is there any issue if a build fails due to a flaky situation and we need to rerun the build. How would we do it? Restart the whole action?